internal/engine/runtimelog/podsource.go

```diff
@@ -0,0 +1,155 @@
+package runtimelog
+
+import (
+	"context"
+	"sync"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/tilt-dev/tilt/internal/k8s"
+	"github.com/tilt-dev/tilt/pkg/logger"
+)
+
+// Helper struct that captures Pod changes and queues up a Reconcile()
```

Comment:
there's definitely some big abstraction here that I'm missing

The controller-runtime has great tools for "if object X creates object Y, then trigger a Reconcile of X every time Y changes".

I feel like we have a common pattern of "object X has a field that references object Y, which means we need a reconcile every time Y change", and then need lots of custom mapping code to do that efficiently.

not sure what the right answer is yet...

Reply:
I get the sense that we might be working against the desired pattern here - since owner references are actually stored on the child, that's how it can map back efficiently on the `.Owns()` (which is more or less syntactic sugar over `.Watches()`).

The Kubebuilder book does have an example of adding an index on a field for client lookups, so I guess that _could_ be a possibility - index your pod field and then in the map func when the pod changes, look up the `PodLogStream`s using the index and emit the requests based off hits.

Reply:
it's even more complicated than that, because the Pod comes from a different API server than the PodLogStream

but I think you're right that the long-term goal should be to have something like Owns() that handles all of this generically....there's nothing really Pod-specific here.